android/src/main/java/com/sentiance/react/bridge/RNSentianceModule.java

```
@@ -277,14 +285,14 @@ public void getVersion(final Promise promise) {
 
   @ReactMethod
   @SuppressWarnings("unused")
-  public void isTripOngoing(String typeParam, final Promise promise) {
+  public void isTripOngoing(Integer typeParam, final Promise promise) {
```

A couple of thing here:

- It makes more sense to keep this a string and change the iOS parameter. The reason is an integer is completely meaningless, whereas something like "sdk" or "external" is self explanatory. Additionally, looking at the format of  trip type we return in user activity changes (using `convertTripType`), we use `TRIP_TYPE_EXTERNAL` for example. It would make sense to align with these.
- This change is not backwards compatible. If possible, I would keep the existing method signatures and overload the method to support the new input type. The existing method can delegate the call to the new one.

---

@sebouh00 
We cannot have overloaded methods in react module, so its either having method with different name or break backward compatibility, I'm more inclined towards breaking backward compatibility as its not a big change.

---

Can we avoid breaking apps if we change `typeParam` to `Object` and `NSObject`, then handle both types accordingly? On iOS, integers can be converted to the new string type name. On Android, old string type names (e.g. "sdk") can be converted to new string type names (e.g. "TRIP_TYPE_SDK"). We update or docs to specify the use of the new types only, but still support older types.

Otherwise, we release a new major version with this method changed.

---

This makes sense!

---

Changed the type back to String with enum values: `TRIP_TYPE_SDK` and `TRIP_TYPE_EXTERNAL`. `sdk` and `external` are still backward supported inside the Android bridge.

For iOS, I did some readings on the supported argument type. `NSObject` is not a valid argument type in which [ReactNative's lib](https://reactnative.dev/docs/native-modules-ios#argument-types) can help convert from a JSON value. So, there is no way that `isTripOngoing` can be invoked with both `Number` and `String` types.

The integration with our current release must be done is this way:
```
if (Platform.OS === 'iOS') {
  RNSentiance.isTripOngoing(1);
} else {
  RNSentiance.isTripOngoing('sdk');
}
```
It already looks abnormal but we haven't received any complaints from clients. I assume there is no live usage of this method and it is safe to break the backward compatibility for this specific case?

Otherwise, we have to use another function name and typing it.

---

@countcain This change is fine. I would play it safe and roll it out with a major version bump. For all we know, someone could be passing integers to this method on both iOS and Android, and it's been silently failing on Android. This change could potentially crash their iOS app.

There's no reason to shy away from a major release if we properly justify the need and clarify the breaking change in the release notes.